### PR TITLE
Make sure this is always one activity and activity section

### PR DESCRIPTION
**Pull request by dmcavoy**
This will make sure there is always at least a blank activity with an activity section in the activities editor so that someone creating a lesson can start adding things right away. In addition when you are editing activities we will always keep one activity section available in every activity and one activity available to edit.

one-activity

## Links

- [jira](https://codedotorg.atlassian.net/browse/PLAT-382)

## Testing story

Added test for removing activity and activity sections in redux that check that a blank activity or activity section is maintained if all of the existing activity/activity sections were to be deleted.

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/sites/studio/pages/lessons/edit.js [hunk 1]
```diff
@@ -53,6 +56,10 @@ $(document).ready(function() {
     });
   });
 
+  if (activities.length === 0) {
+    activities = emptyActivity;
```
Comment (davidsbailey) [inline apps/src/sites/studio/pages/lessons/edit.js]: this looks like a type mismatch -- activities is an array but emptyActivity is a single activity. how about `activities.push(emptyActivity)`? That'd let you switch `let activity` back to `const` earlier, too.

**Review by davidsbailey — APPROVED** [review]
Looks great!

**Comment by davidsbailey** [comment]
Also please feel free to merge this ahead of https://github.com/code-dot-org/code-dot-org/pull/37178 -- it should be easy enough for me to modify that PR once this one is in.

File: apps/src/sites/studio/pages/lessons/edit.js (hunk 1 above)
Comment (dmcavoy) [reply apps/src/sites/studio/pages/lessons/edit.js]: Nice catch. I wonder why proptype validation in the init of activitiesEditorRedux didn't catch this

File: apps/test/unit/lib/levelbuilder/lesson-editor/activitiesEditorReduxTest.js [hunk 2]
```diff
@@ -310,6 +321,18 @@ describe('activitiesEditorRedux reducer tests', () => {
         assert.deepEqual(expectedState, state.activities);
       });
 
+      it('remove last activity section in activity', () => {
+        let state = reducer(initialState, removeActivitySection(1, 1));
+        assert.deepEqual(1, state.activities[0].activitySections.length);
+
+        state = reducer(state, removeActivitySection(1, 1));
+
+        let expectedState = _.cloneDeep(initialActivities);
```
Comment (bethanyaconnor) [inline apps/test/unit/lib/levelbuilder/lesson-editor/activitiesEditorReduxTest.js:330]: Is this ever used in this test?

Comment (dmcavoy) [reply apps/test/unit/lib/levelbuilder/lesson-editor/activitiesEditorReduxTest.js:330]: Good catch! I had planned to check the state against expected state and it looks like I forgot.

**Review by bethanyaconnor — APPROVED** [review]

**Pull request merged**